### PR TITLE
Use pom.xml to describe the packaged plug-in

### PR DESCRIPTION
**Pull request by seanly**
See #75 

Local tests verify the effect

image

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+      <version>0.10</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>pubsub-light</artifactId>
+      <version>1.16</version>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>pipeline-utility-steps</artifactId>
```
Comment (JohnNiang) [inline pom.xml:662]: IMO, adding this plugin to here should be appeared in another Pull Request. Because this PR only refactors the building ways, and should not add any new plugins, so that we can safely verify this PR.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (LinuxSuRen) [reply pom.xml:662]: +1

Normally, we only do one thing in a single PR.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+    <version>4.3</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>io.github.ks-jenkins</groupId>
+  <artifactId>ks-jenkins</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>KubeSphere Jenkins</name>
+  <description>Jenkins formula for KubeSphere.</description>
+  <url>https://kubesphere.github.io/ks-jenkins/</url>
```
Comment (LinuxSuRen) [inline pom.xml:16]: I'm wondering if we can use `https://github.com/kubesphere/ks-jenkins` instead of the below one.

```shell
➜  ~ curl -q https://kubesphere.github.io/ks-jenkins
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

File: formula.yaml [hunk 3]
```diff
@@ -11,606 +11,16 @@ buildSettings:
...
+  pomIgnoreRoot: true
+  pomIncludeWar: true
 systemProperties:
-  { hudson.security.csrf.DefaultCrumbIssuer.EXCLUDE_SESSION_ID: "true" }
+  hudson.security.csrf.DefaultCrumbIssuer.EXCLUDE_SESSION_ID: "true"
+  file.encoding: UTF-8
+  sun.jnu.encoding: UTF-8 
+  jenkins.install.runSetupWizard: "false"
+  org.apache.commons.jelly.tags.fmt.timeZone: Asia/Shanghai
+  user.timezone: Asia/Shanghai
```
Comment (LinuxSuRen) [inline formula.yaml:23]: These settings might not be friendly enough for other users who are not in China.

It could be an option of [the Helm chart](https://github.com/kubesphere-sigs/ks-devops-helm-chart).

File: formula.yaml [hunk 4]
```diff
@@ -11,606 +11,16 @@ buildSettings:
...
-      version: "0.10"
+  pom: pom.xml
+  pomIgnoreRoot: true
+  pomIncludeWar: true
 systemProperties:
-  { hudson.security.csrf.DefaultCrumbIssuer.EXCLUDE_SESSION_ID: "true" }
+  hudson.security.csrf.DefaultCrumbIssuer.EXCLUDE_SESSION_ID: "true"
+  file.encoding: UTF-8
+  sun.jnu.encoding: UTF-8 
+  jenkins.install.runSetupWizard: "false"
```
Comment (LinuxSuRen) [inline formula.yaml:21]: For ks-devops, it's good enough. But in other cases, users might wish to keep it be default.

And we already set it be false at [here](https://github.com/kubesphere-sigs/ks-devops-helm-chart/blob/646534dbbe6ecebaf990e3cd587ed77ec70f5ce5/charts/ks-devops/charts/jenkins/values.yaml#L39)

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
hi @seanly , this is a great try. Thanks for your time. I left some comments, please take look it.
